core/src/main/kotlin/io/fritz2/remote/remote.kt

```diff
@@ -2,32 +2,196 @@ package io.fritz2.remote
+ * By calling one of the executing methods like [get] or [post] a specific request is built from the template and send to the server.
+ *
+ * @property baseUrl the common base of all urls that you want to call using this template
+ * @property errorHandler a common error handler for all requests you will send using this template. By default this just logs the error to the console.
+ */
+class RequestTemplate(val baseUrl : String, val errorHandler: (FetchException) -> Unit = loggingErrorHandler) {
+    private var method: String? = undefined
+    private var headers: Headers? = undefined
+    private var body: String? = undefined
+    private var referrer: String? = undefined
+    private var referrerPolicy: dynamic = undefined
+    private var mode: RequestMode? = undefined
+    private var credentials: RequestCredentials? = undefined
+    private var cache: RequestCache? = undefined
+    private var redirect: RequestRedirect? = undefined
+    private var integrity: String? = undefined
+    private var keepalive: Boolean? = undefined
+    private var window: Any? = undefined
+
+    /**
+     * builts a request, sends it to the server, awaits the response (async), creates a flow of it and attaches the defined errorHandler
+     *
+     * @param url function do derive the url (so you can use baseUrl)
+     * @param init an instance of [RequestInit] defining the attributes of the request
+     */
+    @ExperimentalCoroutinesApi
+    inline fun execute(crossinline url: RequestTemplate.() -> String, init: RequestInit): Flow<Response> = flow {
+        val response = kotlin.browser.window.fetch(url(), init).await()
+
+        if (response.ok) emit(response)
+        else throw FetchException(response.status, response.text().await())
+    }.onError(errorHandler)
+
+    /**
+     * builds a [RequestInit] without a body from the template using [method]
+     *
+     * @param method the http method to use (GET, POST, etc.)
+     */
+    fun buildInit(method: String) = RequestInit(
+        method = method,
+        headers = headers,
+        referrer = referrer,
+        referrerPolicy = referrerPolicy,
+        mode = mode,
+        credentials = credentials,
+        cache = cache,
+        redirect = redirect,
+        integrity = integrity,
+        keepalive = keepalive,
+        window = window
+    )
+
+    /**
+     * builds a [RequestInit] with a body from the template using [method]
+     *
+     * @param method the http method to use (GET, POST, etc.)
+     * @param body content of the request
+     */
+    fun buildInit(method: String, body: String) = RequestInit(
+        method = method,
+        headers = headers,
+        referrer = referrer,
+        referrerPolicy = referrerPolicy,
+        mode = mode,
+        credentials = credentials,
+        cache = cache,
+        redirect = redirect,
+        integrity = integrity,
+        keepalive = keepalive,
+        window = window
+    )
```

Comment:
Why these functions are public? They are a bit confusing when using the API and I think only need internal!?